Let the users change layouts for modules that have nested names for. ex "Example\MyModule"

emcode opened the pull request:
Sometimes it is useful to have few modules for some project grouped under one namespace:

MyProject\System
MyProject\Page
MyProject\User

Currently EdpModuleLayouts does not work for such module names:

array(
    'module_layouts' => array(
        'MyProject\System' => 'layout/system-layout', // <--- this won't work
        'MyProject\Page' => 'layout/page-layout', // <--- this won't work either
    ),
);

I've fixed that.

emcode committed:
Added support for more complex module names like "Example\ModuleName"

EvanDotPro commented:
This assumes they will always have a `Controller` sub-namespace... While it might be true in most cases, I'm not sure we should hard-code such an assumption. I think there is a cleaner way this could be implemented. Basically count the number of namespace identifiers in the module specified in the config, and set a substring offset based on that.

emcode committed:
Merge remote-tracking branch 'upstream/master'

emcode committed:
Changed the way of matching namespaces of Module and Controller

- we do not rely on \\Controller subnamespace anymore
- we have more freedom when specyfying module => layout mapping
  as only beginning of namespace have to match
- remember to set most specific aliases as first, under
  module_layouts config key

emcode commented:
Hey, can you look at this last commit? I think i solved this, but maybe you have some better idea, any comments or suggestions appreciated.

ixtec-zz commented:
This here is quicker and simpler, use the classname as key, and first check if there is a match for it, before testing for the namespace as key, just insert these few lines:

```
        if (isset($config['module_layouts'][$controllerClass])) {
            $controller->layout($config['module_layouts'][$controllerClass]);
            return;
        }
```
